hls4ml/model/optimizer/passes/hgq_proxy_model.py

```diff
@@ -0,0 +1,151 @@
+import re
+from warnings import warn
+
+from hls4ml.backends.fpga.fpga_types import NamedType
+from hls4ml.model.layers import Layer, register_layer
+from hls4ml.model.optimizer import OptimizerPass, register_pass
+from hls4ml.model.types import FixedPrecisionType, UnspecifiedPrecisionType, WeightVariable
+
+re_purge_prefix = re.compile(r'(?<!\w)(?:ap_|ac_)', re.IGNORECASE)
+re_parse_fixed = re.compile(r'\s*(u?)fixed<([^>]+)>\s*', re.IGNORECASE)
+
+
+class FixedPointQuantizer(Layer):
+    def initialize(self):
+        inp = self.get_input_variable()
+        shape = inp.shape
+        dims = inp.dim_names
+        self.add_output_variable(shape, dims)
+        self.set_attr('n_in', self.get_input_variable().size())
+        self.overrides = self.attributes['overrides']
+        self.fusible = self.attributes['fusible']
+        self.SAT, self.RND = self.attributes['SAT'], self.attributes['RND']
+        self.mask_kbi = self.attributes.get('mask_kbi', None)
+
+
+class UnaryLUT(Layer):
+    def initialize(self):
+        inp = self.get_input_variable()
+        shape = inp.shape
+        dims = inp.dim_names
+        self.add_output_variable(shape, dims)
+        self.set_attr('n_in', inp.size())
+        self.table = self.attributes['table']
+        self.table_size = self.attributes['table_size']
+
+        table_t = to_hls4ml_fixed(self.attributes['table_t'])
+        self.add_weights_variable(name='table', var_name='table{index}', precision=table_t, data=self.table)
+
+
+def to_hls4ml_fixed(fixed: str):
```

Comment:
Aren't there standard already-used functions that implement this? I am worried about ac_fixed vs ac_fixed differences here.